### ISSUE TITLE
Compress extracted css

### ISSUE THREAD

**Issue by katyo** [kryops/rollup-plugin-gzip#2]
I use `postcss` plugin (or sass plugin sometimes) with `extract` option enabled to get separate css file.

When I add this css file using `additional` option, it doesn't works as expected:
```js
gzip({
            algorithm: 'zopfli',
            options: {
                level: 9,
                numiterations: 10
            },
            additional: [
                join(distdir, `client_${version}.min.css`)
            ],
        }),
```

```
kayo@debi:$ rollup -c
rollup-plugin-gzip: Error reading file dist/client_0.0.3.min.css
created dist/client_0.0.3.min.js in 3.4s
kayo@debi:$ ls -al dist/client_0.0.3.min.css
-rw-r--r-- 1 kayo kayo dist/client_0.0.3.min.css
```

It seems that compression runs before file actually extracted.

**Comment by kryops** [comment]
Hi @katyo 

thanks for your report!
This seems to be a regression after rollup changed the behavior of how it handles the execution order of plugin hooks. I just opened the issue linked above to clarify what the intended way of handling this case is.

In the meantime, you could try to downgrade rollup to 0.58.

Alternatively, you might try and experiment with the source of this plugin and see if adding a timeout to the compression function would solve the problem for your case:

```
function gzipCompressFile(file, algorithm, options, minSize) {
    return new Promise(resolve => {
        setTimeout(() => {
            fs.stat(file, (err, stats) => {
                if(err) {
                    console.error('rollup-plugin-gzip: Error reading file ' + file);
                    resolve();
                    return;
                }

                if(minSize && minSize > stats.size) {
                    resolve();
                }
                else {
                    const compressor = (algorithm === 'zopfli')
                        ? require('node-zopfli')
                        : require('zlib');

                    fs.createReadStream(file)
                        .pipe(compressor.createGzip(options))
                        .pipe(fs.createWriteStream(file + '.gz'))
                        .on('close', () => resolve());
                }
            });
        }, 5000);
    });
}
```

Maybe then I could add some sort of  `delay` option to this plugin as a (ugly) workaround.

**Comment by katyo** [comment]
Yes. This works as expected with rollup 0.58

**Comment by kryops** [comment]
Hi @katyo 
could you try the new version v1.4.0 of this plugin together with rollup v0.62.0 for your use case?
I tried adding a timeout before compressing the assets for rollup versions affected by the regression. It's set to 5000ms by default, but you can change it using the `delay` option, depending on how long the postcss plugin takes to write its file.

**Comment by kryops** [comment]
Hi @katyo,
I just published v2.0.0 of this plugin as a complete rewrite using the new rollup 0.60 API. Unfortunately, this still has to use the `delay` workaround for additional files written from plugins that use the old API. I also decided to drop the built-in `node-zopfli` support.

For your use case, you will have to change your configuration to something like this when upgrading:

```js
gzip({
	customCompression: content => new Promise((resolve, reject) => {
		require('node-zopfli').gzip(
			content,
			{
				level: 9,
				numiterations: 10
			},
			(err, compressedContent) => {
				if (err) reject('zopfli error');
				else resolve(compressedContent);
			}
		);
	}),
	additionalFiles: [
		join(distdir, `client_${version}.min.css`)
	],
	additionalFilesDelay: 2000 // depending on how long your CSS takes to process
})
```

**Comment by kryops** [comment]
Closing due to inactivity
